src/coreclr/jit/instrsloongarch64.h

```diff
@@ -484,6 +484,1520 @@ INST(fld_d,         "fld.d",          LD,   0x2b800000)
+INST(vfcmp_sne_s,       "vfcmp.sne.s",      0,  0x0c588000)
+INST(vfcmp_sor_s,       "vfcmp.sor.s",      0,  0x0c5a8000)
+INST(vfcmp_sune_s,      "vfcmp.sune.s",     0,  0x0c5c8000)
+INST(vfcmp_caf_d,       "vfcmp.caf.d",      0,  0x0c600000)
+INST(vfcmp_cun_d,       "vfcmp.cun.d",      0,  0x0c640000)
+INST(vfcmp_ceq_d,       "vfcmp.ceq.d",      0,  0x0c620000)
+INST(vfcmp_cueq_d,      "vfcmp.cueq.d",     0,  0x0c660000)
+INST(vfcmp_clt_d,       "vfcmp.clt.d",      0,  0x0c610000)
+INST(vfcmp_cult_d,      "vfcmp.cult.d",     0,  0x0c650000)
+INST(vfcmp_cle_d,       "vfcmp.cle.d",      0,  0x0c630000)
+INST(vfcmp_cule_d,      "vfcmp.cule.d",     0,  0x0c670000)
+INST(vfcmp_cne_d,       "vfcmp.cne.d",      0,  0x0c680000)
+INST(vfcmp_cor_d,       "vfcmp.cor.d",      0,  0x0c6a0000)
+INST(vfcmp_cune_d,      "vfcmp.cune.d",     0,  0x0c6c0000)
+INST(vfcmp_saf_d,       "vfcmp.saf.d",      0,  0x0c608000)
+INST(vfcmp_sun_d,       "vfcmp.sun.d",      0,  0x0c648000)
+INST(vfcmp_seq_d,       "vfcmp.seq.d",      0,  0x0c628000)
+INST(vfcmp_sueq_d,      "vfcmp.sueq.d",     0,  0x0c668000)
+INST(vfcmp_slt_d,       "vfcmp.slt.d",      0,  0x0c618000)
+INST(vfcmp_sult_d,      "vfcmp.sult.d",     0,  0x0c658000)
+INST(vfcmp_sle_d,       "vfcmp.sle.d",      0,  0x0c638000)
+INST(vfcmp_sule_d,      "vfcmp.sule.d",     0,  0x0c678000)
+INST(vfcmp_sne_d,       "vfcmp.sne.d",      0,  0x0c688000)
+INST(vfcmp_sor_d,       "vfcmp.sor.d",      0,  0x0c6a8000)
+INST(vfcmp_sune_d,      "vfcmp.sune.d",     0,  0x0c6c8000)
+INST(vseq_b,            "vseq.b",           0,  0x70000000)
+INST(vseq_h,            "vseq.h",           0,  0x70008000)
+INST(vseq_w,            "vseq.w",           0,  0x70010000)
+INST(vseq_d,            "vseq.d",           0,  0x70018000)
+INST(vsle_b,            "vsle.b",           0,  0x70020000)
+INST(vsle_h,            "vsle.h",           0,  0x70028000)
+INST(vsle_w,            "vsle.w",           0,  0x70030000)
+INST(vsle_d,            "vsle.d",           0,  0x70038000)
+INST(vsle_bu,           "vsle.bu",          0,  0x70040000)
+INST(vsle_hu,           "vsle.hu",          0,  0x70048000)
+INST(vsle_wu,           "vsle.wu",          0,  0x70050000)
+INST(vsle_du,           "vsle.du",          0,  0x70058000)
+INST(vslt_b,            "vslt.b",           0,  0x70060000)
+INST(vslt_h,            "vslt.h",           0,  0x70068000)
+INST(vslt_w,            "vslt.w",           0,  0x70070000)
+INST(vslt_d,            "vslt.d",           0,  0x70078000)
+INST(vslt_bu,           "vslt.bu",          0,  0x70080000)
+INST(vslt_hu,           "vslt.hu",          0,  0x70088000)
+INST(vslt_wu,           "vslt.wu",          0,  0x70090000)
+INST(vslt_du,           "vslt.du",          0,  0x70098000)
+INST(vadd_b,            "vadd.b",           0,  0x700a0000)
+INST(vadd_h,            "vadd.h",           0,  0x700a8000)
+INST(vadd_w,            "vadd.w",           0,  0x700b0000)
+INST(vadd_d,            "vadd.d",           0,  0x700b8000)
```

Comment:
I presume each of these instructions is a variant of `vadd` with a different SIMD lane size. Why did you decide to implement them all as separate instructions instead of a single INS_vadd instruction with a lane size "option", like other platforms do? One result of the way it is done here is a lot of code like:
```
(INS_vreplgr2vr_b <= ins) && (ins <= INS_vreplgr2vr_d)
```
instead of a simple:
```
(ins == INS_vreplgr2vr)
```

And, every single SIMD instruction needs to be listed 4 times.

Reply:
I see several concerns with having so many ranges in the code:
1. It's difficult when reading the code to understand the set of instructions in any given range.
2. It makes it impossible to rearrange the instruction list.

It some cases (many cases?) maybe it would be useful to have named small helper functions that make it more obvious?

Reply:
> I presume each of these instructions is a variant of `vadd` with a different SIMD lane size.

It's `b` for byte, `h` for halfword, `w` for word and `d` for doubleword AFAIR.

Reply:
> > I presume each of these instructions is a variant of `vadd` with a different SIMD lane size.
> 
> It's `b` for byte, `h` for halfword, `w` for word and `d` for doubleword AFAIR.

Yes, you are right.

Reply:
> I presume each of these instructions is a variant of `vadd` with a different SIMD lane size. Why did you decide to implement them all as separate instructions instead of a single INS_vadd instruction with a lane size "option", like other platforms do? One result of the way it is done here is a lot of code like:
> ```
> (INS_vreplgr2vr_b <= ins) && (ins <= INS_vreplgr2vr_d)
> ```

Hi, @BruceForstall 
First of all, thanks for your reviewing.
Even we use a single SIMD instruction `vadd` to express the four `vadd_b/h/w/d`, here the `(INS_vreplgr2vr_b <= ins) && (ins <= INS_vreplgr2vr_d)`  is the same, because we encode the simlar type instructions together not only the `vadd_b/h/w/d`, that is here using `vadd` or `vadd_b/h/w/d` is the same.

Besides, not all SIMD instructions are `b/h/w/d`, using the `b/h/w/d` separately is more exact and there is no need to add size "option" to avoid extra encoding or decoding.

> instead of a simple:
> 
> ```
> (ins == INS_vreplgr2vr)
> ```
> 
> And, every single SIMD instruction needs to be listed 4 times.